packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -23,6 +24,63 @@ export interface IService extends IResource {
   readonly serviceArn: string;
 }
 
+/**
+ * Properties for ECS container targets.
+ */
+export interface TargetProps {
+  /**
+   * Name of the container.
+   */
+  readonly containerName: string,
+
+  /**
+   * Container targets of the container.
+   */
+  readonly containerTargets: ContainerTargetProps[]
+}
+
+/**
+ * Properties for registering a container target.
+ */
+export interface ContainerTargetProps {
+  /**
+   * Container port number to be registered.
+   */
+  readonly containerPort: number,
+
+  /**
+   * Protocol of the container port.
+   *
+   * @default Protocol.TCP
+   */
+  readonly protocol?: Protocol,
+
+  /**
+   * Setting to create and attach target groups.
+   */
+  readonly targetGroups: TargetGroupProps [],
```

Comment:
I would expect a property named `targetGroups` to take `ITargetGroup` objects, not *properties* to *create new* target  groups.